--- a/libpcre.pc.in
+++ b/libpcre.pc.in
@@ -1,6 +1,6 @@
 # Package Information for pkg-config
 
-prefix=@prefix@
+prefix=
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
@@ -8,6 +8,6 @@
 Name: libpcre
 Description: PCRE - Perl compatible regular expressions C library with 8 bit character support
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lpcre
+Libs: ${libdir}/libpcre.a
 Libs.private: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@
 Cflags: -I${includedir} @PCRE_STATIC_CFLAG@
