--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2966,7 +2966,7 @@
 endif()

 find_package(Git)
-if(Git_FOUND)
+if(0)
   execute_process(COMMAND
     "${GIT_EXECUTABLE}" describe --always --tags --long
     WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}"
@@ -2975,7 +2975,7 @@
     ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
 else()
   set(GIT_REVISION_STATUS 1)
-  set(GIT_REVISION "")
+  set(SDL_REVISION "SDL-release-2.26.0-0-g0bfeed061")
 endif()

 if(SDL_SOURCE_VERSION)
