--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2643,7 +2643,7 @@
 
 # Compat helpers for the configuration files
 find_package(Git)
-if(Git_FOUND)
+if(0)
   execute_process(COMMAND
     "${GIT_EXECUTABLE}" remote get-url origin
     WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
@@ -2664,7 +2664,7 @@
     set(SDL_REVISION "")
   endif()
 else()
-  set(SDL_REVISION "")
+  set(SDL_REVISION "https://github.com/libsdl-org/SDL.git@53dea9830964eee8b5c2a7ee0a65d6e268dc78a1")
 endif()
 
 configure_file("${SDL2_SOURCE_DIR}/include/SDL_revision.h.cmake"
