--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2451,7 +2451,7 @@
 
 # Compat helpers for the configuration files
 find_package(Git)
-if(Git_FOUND)
+if(0)
   execute_process(COMMAND
     "${GIT_EXECUTABLE}" remote get-url origin
     WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
@@ -2472,7 +2472,7 @@
     set(SDL_REVISION "")
   endif()
 else()
-  set(SDL_REVISION "")
+  set(SDL_REVISION "https://github.com/libsdl-org/SDL.git@2e9821423a237a1206e3c09020778faacfe430be")
 endif()
 
 configure_file("${SDL2_SOURCE_DIR}/include/SDL_revision.h.cmake"
