--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1902,7 +1902,7 @@
           #import <Foundation/Foundation.h>
           #import <CoreHaptics/CoreHaptics.h>
           int main() { return 0; }" HAVE_FRAMEWORK_COREHAPTICS)
-      if(HAVE_FRAMEWORK_GAMECONTROLLER AND HAVE_FRAMEWORK_COREHAPTICS)
+      if(0)
         # Only enable MFI if we also have CoreHaptics to ensure rumble works
         set(SDL_JOYSTICK_MFI 1)
         set(SDL_FRAMEWORK_GAMECONTROLLER 1)
