--- a/configure
+++ b/configure
@@ -906,7 +906,6 @@
 /^ARFLAGS *=/s#=.*#=$ARFLAGS#
 /^RANLIB *=/s#=.*#=$RANLIB#
 /^EXE *=/s#=.*#=$EXE#
-/^prefix *=/s#=.*#=$prefix#
 /^exec_prefix *=/s#=.*#=$exec_prefix#
 /^libdir *=/s#=.*#=$libdir#
 /^sharedlibdir *=/s#=.*#=$sharedlibdir#
--- a/zlib.pc.in
+++ b/zlib.pc.in
@@ -1,4 +1,4 @@
-prefix=@prefix@
+prefix=
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 sharedlibdir=@sharedlibdir@
@@ -9,5 +9,5 @@
 Version: @VERSION@
 
 Requires:
-Libs: -L${libdir} -L${sharedlibdir} -lz
+Libs: ${libdir}/libz.a
 Cflags: -I${includedir}
